packages/core/admin/utils/watch-admin-files.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const path = require('path');
+const fs = require('fs-extra');
+const chokidar = require('chokidar');
+const { isUsingTypeScript } = require('@strapi/typescript-utils');
+
+/**
+ * Listen to files change and copy the changed files in the .cache/admin folder
+ * when using the dev mode
+ * @param {string} dir
+ */
+async function watchAdminFiles(dir) {
+  const useTypeScript = await isUsingTypeScript(path.join(dir, 'src', 'admin'), 'tsconfig.json');
+
+  const cacheDir = path.join(dir, '.cache');
+  const targetExtensionFile = useTypeScript ? 'app.tsx' : 'app.js';
```

Comment:
could also be app.ts with chokidar we can use globs that would probably be simpler. and I don't think we should check if it's ts or not here

Reply:
it's tsx? in the rest of the code the filename isn't .tsx or .ts but also .ts If I read the rest of the code correctly